app/utils/callback/lava.py

```diff
@@ -594,12 +448,194 @@ def _translate_log_end_lines(results, translate_map):
                 result['log_end_line'] = new_log_end_line
 
 
-def _prepare_log(log):
-    log = yaml.load(log, Loader=yaml.CLoader)
-    for i, log_line in enumerate(log, 1):
-        log_line['msg'] = unicode(log_line['msg'])
-        log_line['org_num'] = i
-    return log
+class LavaCallback(object):
+    META_DATA_MAP_TEST = {
+        models.ARCHITECTURE_KEY: "job.arch",
+        models.BUILD_ENVIRONMENT_KEY: "job.build_environment",
+        models.DEFCONFIG_KEY: "kernel.defconfig",
+        models.DEFCONFIG_FULL_KEY: "kernel.defconfig_full",
+        models.DEVICE_TYPE_KEY: "device.type",
+        models.DTB_KEY: "platform.dtb",
+        models.ENDIANNESS_KEY: "kernel.endian",
+        models.FILE_SERVER_RESOURCE_KEY: "job.file_server_resource",
+        models.GIT_BRANCH_KEY: "git.branch",
+        models.GIT_COMMIT_KEY: "git.commit",
+        models.GIT_DESCRIBE_KEY: "git.describe",
+        models.GIT_URL_KEY: "git.url",
+        models.INITRD_KEY: "job.initrd_url",
+        models.JOB_KEY: "kernel.tree",
+        models.KERNEL_KEY: "kernel.version",
+        models.KERNEL_IMAGE_KEY: "job.kernel_image",
+        models.MACH_KEY: "platform.mach",
+        models.PLAN_KEY: "test.plan",
+        models.PLAN_VARIANT_KEY: "test.plan_variant",
+    }
+
+    @classmethod
+    def process_callback(cls, job_data, definition_meta, lab_name,
+                         base_path=utils.BASE_PATH):
+        utils.LOG.info("Processing LAVA test data: job {} from {}".format(
+            job_data["id"], lab_name))
+
+        if job_data.get("status") not in (COMPLETE, INCOMPLETE):
+            utils.LOG.warning("Skipping LAVA job due to unsupported status: "
+                              "{}".format(job_data["status_string"]))
+            return None
+        return cls(job_data, definition_meta, lab_name, base_path)
+
+    def __init__(self, job_data, definition_meta, lab_name,
+                 base_path):
```

Comment:
No need for a line wrap here, it should fit in 79 characters.

Reply:
unwrapped